hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -272,7 +272,7 @@ private void updateState() {
     } else if (localTarget == null) {
       // block info available, HA status not expected
       LOG.debug(
-          "Reporting non-HA namenode as operational: " + getNamenodeDesc());
+          "Reporting non-HA namenode as operational: {}", getNamenodeDesc());
```

Comment:
I think this statement really needs wrapped in an if `(LOG.isDebugEnabled())`. The reason is that the method call `getNamenodeDesc()` needs to be evaluated whether we log or not, so its result can be passed into the `LOG.debug()` method. Inside `LOG.debug`, it will skip the logging, but by then, we have already formed the string inside `getNamenodeDesc()` and never use it.

If we are passing an object into `LOG.debug`, this change would be the correct thing to do, as toString() will only get called on the object if the log message is created. However when we pass a method call into the method, it needs to be evaluated first AFAIK.

Reply:
Note there is another debug message just above this one which looks similar and could be wrapped in a if LOG.isDebugEnabled at 270 for the same reason.

Reply:
Modified, please help review again. Thanks!

Reply:
The latest change looks good. Thanks for changing both the log lines.